File: beginner/__2000__/2176/2176.cpp
#include <iostream>

using namespace std;

int main(){

    string input;
    cin >> input;

    int bits_1 = 0;
    for(int i = 0; i < input.size(); i++){
        char bit = input[i];
        bits_1 += (bit == "1" ? 1 : 0);
    }

    if(bits_1 % 2 == 0){
        input += "0";
    } else {
        input += "1";
    }

    cout << input << endl;
    
    return 0;

}

File: beginner/__2000__/2176/2176.lua
local segment = io.read()

local bits_1 = 0
for i = 1, #segment do 
    local bit = segment:sub(i, i)
    if bit == "1" then
        bits_1 = bits_1 + 1
    end
end

if bits_1 % 2 == 0 then
    segment = segment .. "0"
else 
    segment = segment .. "1"
end

print(segment)